modules/zivid/captureassistant.py

```diff
@@ -0,0 +1,85 @@
+"""Contains capture assistant functions and classes."""
+from enum import Enum
+
+import _zivid
+import zivid._settings_converter as _settings_converter
+
+
+class AmbientLightFrequency(Enum):  # pylint: disable=too-few-public-methods
+    """Ensure compatibility with the frequency of the ambient light in the scene."""
+
+    hz50 = _zivid.captureassistant.AmbientLightFrequency.hz50
+    hz60 = _zivid.captureassistant.AmbientLightFrequency.hz60
+    none = _zivid.captureassistant.AmbientLightFrequency.none
+
+    def __str__(self):
```

Comment:
Why overload `__str__()`? 
The default implementation from `Enum` would yield e.g. `AmbientLightFrequency.hz50`, which is perhaps even more readable.

Reply:
This could be true. ATM, rationale was to be close to c++ implementation.
This could surely be modified.
@nedrebo your take?

Reply:
Yes, this is just tmp code to make sure the rest could be implemented and tested. How to wrap enums are still undecided. We have the standard requirements about robust code, but I did not spend time on actually coming up with a design. As it is stands now we are leaking pybind11 types `zivid`, that is not something we can merge. pybind11 should not be observable from `zivid` (using "normal" code).

Reply:
In which sense are we leaking pybind11 types?
Is it because
```
hzmode = zivid.captureassistant.AmbientLightFrequency.hz50
print(type(hzmode.value))
```
gives 
> <class '_zivid._zivid.captureassistant.AmbientLightFrequency'>
?

Reply:
We could perhaps do the following:

```
# Hidden mapping to underlying pybind11 type
_AmbientFrequencyChoices = ["hz50", "hz60", "none"]
_AmbientFrequencyMap = {choice: getattr(_zivid.captureassistant.AmbientLightFrequency, choice) 
                        for choice in _AmbientFrequencyChoices}

AmbientLightFrequency = enum.Enum('AmbientLightFrequency', _AmbientFrequencyChoices)

# Visible enum with normal integer underlying type
AmbientLightFrequency = enum.Enum('AmbientLightFrequency', [*_AmbientFrequencyChoices])

class SuggestSettingsParameters: 
    def __init__(self, budget, frequency=None):
        self.__impl = _zivid.captureassistant.SuggestSettingsParameters(
            budget, _AmbientFrequencyMap[frequency.name]
        )

```

This has the following features: 
- completely hides the pybind11 type in the private/"underscore"  dictionary.
- defines the mapping to the pybind11 type in a single place.
- Impossible to mistakenly map e.g. "hz50" to "hz60"

Reply:
As discussed IRL, this could be a good candidate (bit changed @eskaur 's suggestion)
`_ALFEntries = {entry: entry for entry in _zivid.capture_assistant.AmbientLightFrequency.__entries}
_ALFToInternalMap = {entry: getattr(_zivid.capture_assistant.AmbientLightFrequency, entry) for entry in _ALFEntries}

AmbientLightFrequency = Enum('AmbientLightFrequency', _ALFEntries)
setattr(AmbientLightFrequency, '__str__', lambda self: str(self.name))`

Reply:
now in fixup

Reply:
Looks good to me, but I'll let @nedrebo decide if it properly addressed his initial concern.

Reply:
There are some linter issues - will try to fix them